contracts/DomainRegistryV2.sol

```diff
@@ -0,0 +1,310 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.24;
+
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import "solidity-stringutils/src/strings.sol";
+
+/**
+* @title Domain Registry Contract
+* @notice This contract allows users to register and manage top-level domain names
+* @author Developed by Ivan Myasoyedov
+*/
+contract DomainRegistryV2 is OwnableUpgradeable {
```

Comment:
// contract should implement NatSpec in strict order 
// constants should be declared at the top of the contract
// private functions should be declared at the bottom of the contract
/// // - 3 points

Reply:
Done, please check